KG-545 Fix `requestLLMOnlyCallingTools` ignoring tool calls after reasoning message

mltheuser opened the pull request:
## Motivation and Context
Related to [KG-545](https://youtrack.jetbrains.com/issue/KG-545).

Currently, `requestLLMOnlyCallingTools` relies on `executeSingle`, which returns the first message received from the LLM. When using models that output Chain of Thought or "Thinking" blocks (e.g., Nova, Claude) prior to calling a tool, the response sequence is often `[Message.Assistant(Thinking), Message.Tool.Call]`.

As a result:
1. The method returns the "Thinking" text message instead of the expected Tool Call.
2. The actual Tool Call is discarded and never executed or saved to history.

**Changes:**
*   Introduced `requestLLMMultipleOnlyCallingTools()` in `AIAgentLLMSession` to allow retrieving the full list of messages while enforcing `ToolChoice.Required`.
*   Updated `requestLLMOnlyCallingTools` to use this new method. It now persists **all** messages (preserving the reasoning context in the session history) but filters the return value to ensure the caller receives the `Message.Tool.Call`.

## Breaking Changes
None. This is a behavioral fix to ensure the method contract (returning a tool call) is honored when the LLM is "chatty" or provides reasoning.

---

#### Type of the changes
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tests improvement
- [ ] Refactoring

#### Checklist
- [x] The pull request has a description of the proposed change
- [x] I read the [Contributing Guidelines](https://github.com/JetBrains/koog/blob/main/CONTRIBUTING.md) before opening the pull request
- [x] The pull request uses **`develop`** as the base branch
- [x] Tests for the changes have been added
- [x] All new and existing tests passed

##### Additional steps for pull requests adding a new feature
- [x] An issue describing the proposed change exists
- [x] The pull request includes a link to the issue
- [x] The change was discussed and approved in the issue
- [ ] Docs have been added / updated

Commit:
KG-545 Fix `requestLLMOnlyCallingTools` ignoring tool calls after reasoning

EugeneTheDev commented on agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/session/AIAgentLLMWriteSession.kt:
```diff
@@ -392,13 +392,19 @@ public class AIAgentLLMWriteSession internal constructor(
-        return super.requestLLMOnlyCallingTools().also { response -> appendPrompt { message(response) } }
+    override suspend fun requestLLMMultipleOnlyCallingTools(): List<Message.Response> {
+        return super.requestLLMMultipleOnlyCallingTools().also { responses ->
+            appendPrompt {
```
nit: You can simplify it 
```kotlin
appendPrompt {
    messages(responses)
}

```

EugeneTheDev reviewed (approved):
Looks good, thank you. Please check a small comment

kpavlov commented on agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/session/AIAgentLLMSession.kt:
```diff
@@ -164,17 +164,42 @@ public sealed class AIAgentLLMSession(
+     * 4. Filters the result to return the first [Message.Tool.Call].
      *
-     * @return The response from the language model after executing the request with enforced tool usage.
+     * If no tool call is found (e.g., the model refused or asked a question), it falls back to the first message.
```
Let’s also make sure this use case gets tested.

However, I’m not sure we should have a fallback; perhaps we should just fail.

kpavlov commented on agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/agent/session/AIAgentLLMSession.kt:
```diff
@@ -164,17 +164,42 @@ public sealed class AIAgentLLMSession(
             toolChoice = LLMParams.ToolChoice.Required
         }
-        return executeSingle(promptWithOnlyCallingTools, tools)
+        return executeMultiple(promptWithOnlyCallingTools, tools)
```
👍🏻

kpavlov reviewed (commented):
Thank you, @mltheuser
I’m on board with the overall change.

Just one thing: should we handle the case where no tool call is found by failing? I’d lean towards that, but I’m not set in stone.

I’m happy to merge it, as we have tests in place to ensure we’re covering all edge cases.

`requestLLMOnlyCallingTools()`:
1. "If no tool call is found (e.g., the model refused or asked a question), it falls back to the first message."
2. What would happen when the model returns multiple tools

Commit:
KG-545 resolve PR comments

mltheuser commented:
Thanks for the review!

@EugeneTheDev I've applied the simplification using `messages()` as suggested — much cleaner.

@kpavlov Good call on the edge cases. I've updated `requestLLMOnlyCallingTools` to fail explicitly if no tool call is returned, rather than falling back to the text message. I also added new tests to cover that scenario and the multiple tool calls case (it now correctly returns the first one while preserving all context).

Commit:
KG-545 fix ktlint error - shame on me
